src/heap.rs

```diff
@@ -0,0 +1,247 @@
+/*
+General implementation of a heap
+*/
+use bytemuck::{Pod, Zeroable};
+use std::cmp::PartialOrd;
+
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Node<T> {
+    pub v: T,
```

Comment:
What is the purpose of making this a struct? If anything I think you would want to make `Node<K, V>` so you can use `K` as a sorting key. If it's purely `T`, then you definitely don't need `Node`